crates/ra_analysis/src/descriptors.rs

```diff
@@ -218,3 +219,56 @@ fn resolve_submodule(
     }
     (points_to, problem)
 }
+
+#[derive(Debug, Clone)]
+pub struct FnDescriptor {
+    pub name: Option<String>,
```

Comment:
Let's make this non-optional (FnDescriptor without a name is pretty useless), and instead change return type of `new` to `fn new(node: ast::FnDef) -> Option<Self>`?